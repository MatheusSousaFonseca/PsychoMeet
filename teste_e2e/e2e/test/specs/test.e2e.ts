import { expect } from '@wdio/globals'
import LoginPage from '../pageobjects/login.page.js'
import SecurePage from '../pageobjects/secure.page.js'

describe('My Login application', () => {
    it('should login with valid credentials', async () => {
        await LoginPage.open()
        await LoginPage.login('joao@gmail.com', 'senha123')
        await expect(SecurePage.flashAlert).toBeExisting()
        await expect(SecurePage.flashAlert).toHaveText(
            expect.stringContaining('Hist칩rico de Consultas'))
        await expect(SecurePage.flashAlert).toMatchSnapshot('Hist칩rico de Consultas')
    })
})

describe('My Login application', () => {
    it('should not login with invalid password', async () => {
        await LoginPage.open()
        await LoginPage.login('joao@gmail.com', 'senha12345')
        await browser.pause(3000);
        const messageToastr = await LoginPage.getMessageToastr();
        const text = await messageToastr.getText();
        expect(text).toContain('Dados inv치lidos');
        await browser.pause(3000);
    })
    
    it('should not login with invalid email', async () => {
        await LoginPage.open()
        await LoginPage.login('joaooo@gmail.com', 'senha123')
        await browser.pause(3000);
        const messageToastr = await LoginPage.getMessageToastr();
        const text = await messageToastr.getText();
        expect(text).toContain('Dados inv치lidos');
        await browser.pause(3000);
    })
})


describe('My Login application', () => {
    it('should have the submit button disabled when the form is incomplete', async () => {
        await LoginPage.open(); 
        await LoginPage.inputEmail.setValue('joao@gmail.com');
        await LoginPage.inputSenha.setValue('');
        const isButtonDisabled = await LoginPage.btnSubmit.getAttribute('disabled');
        await expect(isButtonDisabled).toBe('true');
    });
});

