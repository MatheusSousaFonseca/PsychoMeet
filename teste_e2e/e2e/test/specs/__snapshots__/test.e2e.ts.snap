// Snapshot v1

exports[`My Login application > should login with valid credentials > Histórico de Consultas 1`] = `"<h2 _ngcontent-ng-c2945389749="" id="titulo" class="card-title">Histórico de Consultas</h2>"`;

exports[`My Login application > should login with valid credentials > flashAlert 1`] = `
"<div data- id="flash" class="flash success">You logged into a secure area!
  <a href="#" class="close">×</a>
</div>"
`;
